stages/org.osbuild.kickstart

```diff
@@ -198,6 +198,10 @@ SCHEMA = r"""
         }
       }
     }]
+  },
+  "text": {
+    "type": "boolean",
```

Comment:
Would it make sense to make this an enum, to make it clear that you can only set one of `text`, `graphical` and `cmdline`?

Reply:
Hey @teg, thanks for your input on this! I like the idea and I'm happy to work on this. I see two ways of going about this (but maybe there are more?):
1. add text,graphical,cmdline as boolean options and use `oneOf` in the schema to limit what can be used together
2. add a new value like `install-mode` or `install-environment` and make that an enum with [text,graphical,cmdline]
So far in the kickstart stage maps all kickstart commands and their options almost directly to schema properties (ostree.gpg is an exception though) which is nice because it means there is not really a need to look into the osbuild schema as it translates pretty much 1:1. This is why I prefer option (1) slightly more, option (2) would mean that the user would have to check the osbuild schema for this option.

I think option (1) can be done the following diff. However one downside of (1) is that the error message from jsonschema seems to be not great when multiple options are selected (e.g. `{'path': 'some-path', 'text': True, 'cmdline': True} is valid under each of {'required': ['cmdline']}, {'required': ['text']}"`. Option (2) most likely yields better errors. Input on what the right tradeoff is is very welcome!
```python3
diff --git a/stages/org.osbuild.kickstart b/stages/org.osbuild.kickstart
index b4a9d100..9c1ac029 100755
--- a/stages/org.osbuild.kickstart
+++ b/stages/org.osbuild.kickstart
@@ -19,6 +19,16 @@ import osbuild.api
 SCHEMA = r"""
 "additionalProperties": false,
 "required": ["path"],
+"oneOf": [
+    {"required": ["text"]},
+    {"required": ["cmdline"]},
+    {"required": ["graphical"]},
+    {"allOf": [
+      {"not": {"required": ["text"]}},
+      {"not": {"required": ["cmdline"]}},
+      {"not": {"required": ["graphical"]}}
+    ]}
+],
 "properties": {
   "path": {
     "type": "string",
@@ -202,6 +212,14 @@ SCHEMA = r"""
   "text": {
     "type": "boolean",
     "description": "Perform the Kickstart installation in text mode"
+  },
+  "cmdline": {
+    "type": "boolean",
+    "description": "Perform the installation in a completely non-interactive command line mode"
+  },
+  "graphical": {
+    "type": "boolean",
+    "description": "Perform the installation in graphical mode. This is the default."
   }
 }
 """
@@ -304,7 +322,7 @@ def make_reboot(options):
     return cmd
 
 
-def main(tree, options):
+def main(tree, options):  # pylint: disable=too-many-branches
     path = options["path"].lstrip("/")
     ostree = options.get("ostree")
 
@@ -346,9 +364,12 @@ def main(tree, options):
     clearpart = make_clearpart(options)
     if clearpart:
         config += [clearpart]
-    text = options.get("text")
-    if text:
+    if options.get("text"):
         config += ["text"]
+    if options.get("cmdline"):
+        config += ["cmdline"]
+    if options.get("graphical"):
+        config += ["graphical"]
 
     reboot = make_reboot(options)
     if reboot:
diff --git a/stages/test/test_kickstart.py b/stages/test/test_kickstart.py
index 3d942882..b771e3a8 100644
--- a/stages/test/test_kickstart.py
+++ b/stages/test/test_kickstart.py
@@ -105,6 +105,8 @@ TEST_INPUT = [
     ({"reboot": {"kexec": True}}, "reboot --kexec"),
     ({"reboot": {"eject": True, "kexec": True}}, "reboot --eject --kexec"),
     ({"text": True}, "text"),
+    ({"cmdline": True}, "cmdline"),
+    ({"graphical": True}, "graphical"),
 ]
 
 
@@ -183,6 +185,11 @@ def test_kickstart_valid(tmp_path, test_input, expected):  # pylint: disable=unu
         ({"reboot": {}}, "{} is not valid under any of the given schemas"),
         ({"reboot": "random-string"}, "'random-string' is not valid "),
         ({"reboot": {"random": "option"}}, "{'random': 'option'} is not valid "),
+        # only one of text/cmdline/graphical can be used at the same time
+        ({"text": True, "cmdline": True, "graphical": True}, " is valid under each of "),
+        ({"text": True, "cmdline": True}, " is valid under each of "),
+        ({"text": True, "graphical": True}, " is valid under each of "),
+        ({"cmdline": True, "graphical": True}, " is valid under each of "),
     ],
 )
 def test_schema_validation_bad_apples(test_data, expected_err):
```

Reply:
I do like the idea of mapping our stage options and their restrictions to the file we're creating, but I think I prefer the simplicity of the schema with an enum.  It doesn't map to the format of the file but it maps to the logic of it.  I don't feel too strongly about it though.

Reply:
Sure, going with the enum works for me. I was thinking `["install_mode": { "enum": ["text", "graphics", "cmdline"]}}` or are there better ideas for `install_mode` ? Alternative ideas: `install_ui`, `install_presentation`, `install_ui_mode` (sorry for the bikeshed but given that we need to live with the name I want to double check without).

Reply:
https://pykickstart.readthedocs.io/en/latest/kickstart-docs.html#graphical-or-text-or-cmdline calls it "display mode" so maybe we can go with that.

Reply:
And the same here: https://github.com/pykickstart/pykickstart/blob/925b5cd211ad66b82d082bb42260353d20130da9/pykickstart/constants.py#L30-L32

Reply:
\o/ `display_mode` it is then. Thank you!

Reply:
Thanks for the detailed analysis! The solution you both came up with makes sense to me too!

Reply:
Updated as discussed and ready for a re-review